lib/App/Sqitch/Engine/mysql.pm

```diff
@@ -171,7 +171,8 @@ has _mysql => (
 
         # Special-case --password, which requires = before the value. O_o
         if (my $pw = $self->password) {
-            push @ret, "--password=$pw";
```

Comment:
The password in `$pw` can be different from the one in the file, so I think this should probably be:

``` diff
 
         # Special-case --password, which requires = before the value. O_o
         if (my $pw = $self->password) {
-            push @ret, "--password=$pw";
+            my $cfgpw = $self->_mycnf->{password} || '';
+            push @ret, "--password=$pw" if $pw ne $cfgpwd;
         }
 
         # Options to keep things quiet.
```

Reply:
I have added this change ;)